src/Php72/Php72.php

```diff
@@ -97,16 +97,33 @@ public static function spl_object_id($object)
 
     public static function sapi_windows_vt100_support($stream, $enable = null)
     {
+        if (!is_resource($stream)) {
+            trigger_error('sapi_windows_vt100_support() expects parameter 1 to be resource, '.gettype($stream).' given', E_USER_WARNING);
+            return false;
+        }
+
+        $meta = stream_get_meta_data($stream);
+
+        if ('STDIO' !== $meta['stream_type']) {
+            $read = null;
+            $write = array($stream);
+            $except = null;
+
+            if (false === @stream_select($read, $write, $except, 0)) {
+                trigger_error('sapi_windows_vt100_support() was not able to analyze the specified stream', E_USER_WARNING);
+            }
+
+            return false;
+        }
+
         // We cannot actually disable vt100 support if it is set
         if (false === $enable || !self::stream_isatty($stream)) {
             return false;
         }
 
-        // The native function does not apply to stdin
-        $meta = array_map('strtolower', stream_get_meta_data($stream));
-        $stdin = 'php://stdin' === $meta['uri'] || 'php://fd/0' === $meta['uri'];
+        $writable = false !== strpbrk($meta['mode'], 'awxc+');
 
```

Comment:
I had to use the `uri` value because `mode` reports whatever values were supplied by the user. For example: `fopen('php://stdin', 'wb')` reports 'wb'

Reply:
I'll compare what native function does then and try to enhance it somehow.

Reply:
The native function checks that the stream is not stdin. From php-src/win32/console.c:

`if (fileno != 0 && !GetNumberOfConsoleInputEvents(handle, &events))`